@xen-orchestra/proxy/src/app/mixins/appliance.js

```diff
@@ -144,11 +121,57 @@ export default class Appliance {
           ],
         },
         updater: {
-          getLocalManifest: () => _withUpdater(_ => _.call('getLocalManifest')),
-          getState: () => callUpdate(),
-          upgrade: () => callUpdate({ upgrade: true }),
+          getLocalManifest: () => using(this.getUpdater(), _ => _.call('getLocalManifest')),
+          getState: () => this._callUpdate(),
+          upgrade: () => this._callUpdate({ upgrade: true }),
         },
       },
     })
   }
+
+  _callUpdate(params) {
+    return using(
+      this.getUpdater(),
+      updater =>
+        new Promise((resolve, reject) => {
+          updater
+            .on('error', reject)
+            .on('notification', ({ method, params }) => {
+              if (method === 'print') {
+                debug('updater.update: ' + params.content)
+              } else if (method === 'end') {
+                resolve(params)
+              } else if (method === 'server-error') {
+                reject(new Error(params.message))
+              } else if (method !== 'connected') {
+                warn('update.update, unhandled message', {
+                  method,
+                  params,
+                })
+              }
+            })
+            .notify('update', params)
+        })
+    )
+  }
+
+  @decorateResult(function (resource) {
+    return debounceResource(resource, this._app.hooks, parseDuration(this._config.resourceDebounce))
+  })
+  @decorateWith(deduped)
+  @decorateWith(disposable)
+  async *getUpdater() {
+    const updater = new JsonRpcWebsocketClient('ws://localhost:9001')
+    await updater.open()
+    try {
+      yield updater
+    } finally {
+      await updater.close().then(Function.prototype)
```

Comment:
I don't understand the reason for this `.then`, let's remove it or comment it :slightly_smiling_face: